golang/x/relay/keeper/handler.go

```diff
@@ -4,17 +4,46 @@ import (
 	"fmt"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/summa-tx/relays/golang/x/relay/types"
 )
 
 // NewHandler returns a handler for "nameservice" type messages.
 func NewHandler(keeper Keeper) sdk.Handler {
 	return func(ctx sdk.Context, msg sdk.Msg) sdk.Result {
 		switch msg := msg.(type) {
-		// case types.MsgSetLink:
-		// 	return handleMsgSetLink(ctx, keeper, msg)
+		case types.MsgIngestHeaderChain:
+			return handleMsgIngestHeaderChain(ctx, keeper, msg)
+		case types.MsgIngestDifficultyChange:
+			return handleMsgIngestDifficultyChange(ctx, keeper, msg)
+		case types.MsgMarkNewHeaviest:
+			return handleMsgMarkNewHeaviest(ctx, keeper, msg)
 		default:
 			errMsg := fmt.Sprintf("Unrecognized nameservice Msg type: %v", msg.Type())
```

Comment:
> nameservice

should say relay. leftover from copy-pasting from elsewhere :rofl:

Reply:
Good catch, I fixed this in a couple other places as well :D